packages/vitest-pool-workers/src/pool/module-fallback.ts

```diff
@@ -75,6 +75,17 @@ function isFile(filePath: string): boolean {
 	}
 }
 
+function isDirectory(filePath: string): boolean {
+	try {
+		return fs.statSync(filePath).isDirectory();
+	} catch (e) {
+		if (isFileNotFoundError(e)) {
+			return false;
+		}
+		throw e;
+	}
+}
+
```

Comment:
We have basically the same code in several other places in the repo - I wonder if we can consolidate these?

[1](https://github.com/cloudflare/workers-sdk/blob/c2bb42fefa402ece691fd4b034a749d2862ec61a/packages/wrangler/src/global-wrangler-config-path.ts/#L6-L13), [2](https://github.com/cloudflare/workers-sdk/blob/c2bb42fefa402ece691fd4b034a749d2862ec61a/packages/wrangler/src/pages/functions/buildPlugin.ts/#L67-L72), [3](https://github.com/cloudflare/workers-sdk/blob/c2bb42fefa402ece691fd4b034a749d2862ec61a/packages/wrangler/src/pages/functions/buildWorker.ts/#L94-L99), [4](https://github.com/cloudflare/workers-sdk/blob/c2bb42fefa402ece691fd4b034a749d2862ec61a/packages/create-cloudflare/src/helpers/files.ts/#L39-L49). [5](https://github.com/cloudflare/workers-sdk/blob/c2bb42fefa402ece691fd4b034a749d2862ec61a/packages/wrangler/src/pages/dev.ts/#L351-L352), [6](https://github.com/cloudflare/workers-sdk/blob/c2bb42fefa402ece691fd4b034a749d2862ec61a/packages/wrangler/src/pages/functions.ts/#L48-L49)

Reply:
For sure within a package (such as the 4 versions in wrangler). I am less bothered about sharing little utilities like this between packages.

Reply:
Feel free to follow up with clean up PRs for that sort of thing.

Reply:
The case for, I would say, is to both standardise and to avoid common errors. This one, for example, will throw an error that TS doesn't pick up:

```
const t = fs.statSync('sdfs').isDirectory();
```

There are other common pitfalls, like `parseInt(couldBeEmptyString, 10) -> Nan`.

One more reason is that we have some nice little gems that might never get discovered because they're hidden away, like [this one from the C3 code base](https://github.com/cloudflare/workers-sdk/blob/c2bb42fefa402ece691fd4b034a749d2862ec61a/packages/create-cloudflare/src/helpers/retry.ts/#L24-L39).

I think a common utility library could make our code more consistent as well as more robust and discoverable.